mcstatus/protocol/connection.py

```diff
@@ -19,10 +19,11 @@
 if TYPE_CHECKING:
     from typing_extensions import SupportsIndex  # Python 3.7 doesn't support this yet.
 
-    BytesConvertable = Union[SupportsIndex, Iterable[SupportsIndex]]
+    # don't supported by <3.10, but it's run in `TYPE_CHECKING` block!
+    BytesConvertable = SupportsIndex | Iterable[SupportsIndex]  # type: ignore
```

Comment:
[Still error on Python 3.9](https://github.com/py-mine/mcstatus/runs/8170794389?check_suite_focus=true#step:11:23). Rolling back this change.

Reply:
We should be able to lock pyright to version 3.10 in pyright's section in `pyproject.toml`

Reply:
We shouldn't actually, this will force `pyright` to ignore `|` even if there is no `__future__.annotations`.

Better will be just report to `pyright`, as it is a bug.

Reply:
Created https://github.com/microsoft/pyright/issues/3961 for this.

Reply:
> We shouldn't actually, this will force `pyright` to ignore `|` even if there is no `__future__.annotations`.
> 
> Better will be just report to `pyright`, as it is a bug.

That would fail on runtime anyway though, we don't need pyright to inform us here. But I do agree that it's probably not the best solution.

Reply:
I propose to left type ignore at this moment, and when pyright will fix the bug, we will remove it.

Reply:
> https://github.com/microsoft/pyright/issues/3961#issuecomment-1250333401

This isn't a bug.

Reply:
Perhaps doing 
```python
BytesConvertable: TypeAlias = "SupportsIndex | Iterable[SupportsIndex]"
```
would make more sense then.

Reply:
However, it works normally in runtime, and using `"` will cause confusions in IDE. For example, if something will be renamed, IDE will think that everything OK. And only type-checker possibly will show the error.

I think `type: ingore` produces fewer confusions.

Reply:
On runtime, invalid annotation like `BytesConevrtable: TypeAlias = NotExistentThing  # type: ignore` will be even worse then. The type ignore will suppress any issues from pyright, including the one about `NotExistentThing` not existing, and since this would be in the `TYPE_CHECKING` block, it won't even run on runtime, meaning no errors will show up.

Yes, by making it a string we won't get runtime exceptions if the class isn't defined, but because we use `__future__.annotations`, we wouldn't get these from annotated function parameters anyway, and it's not an issue to make type aliases act similarly. Type ignore is way more problematic here and I think simply making it a string makes much more sense.

Reply:
Fixed in 8d489fd02cf493277f79d41288bb4118fd4b9645.